code/modules/mob/living/simple_animal/friendly/mule_bot/mule_bot.dm

```diff
@@ -0,0 +1,202 @@
+	if(R)
+		AddComponent(/datum/component/ai_controller, /datum/ai_behavior/mule_bot, R)
+
+/mob/living/simple_animal/mule_bot/update_overlays()
+	. = ..()
+	. += list(hat_overlay,face_overlay,installed_module?.mod_overlay)
+
+
+/mob/living/simple_animal/mule_bot/attackby(obj/item/I, mob/living/user, def_zone)
+	//give it a funny hat, could be turned into an actuel proc though
+	if(istype(I,/obj/item/clothing/head))
+		if(hat)
+			hat.forceMove(src.loc)
+		var/obj/item/clothing/head/new_hat = I
+		I.forceMove(src)
+		user?.temporarilyRemoveItemFromInventory(I)
+		hat = new_hat
+		hat_overlay = mutable_appearance(new_hat.get_worn_icon_file("Human",slot_head_str), new_hat.get_worn_icon_state(slot_head_str), HEAD_LAYER, FLOAT_PLANE)
+		hat_overlay.pixel_y -= -5
+		update_icon()
+		return
+	if(istype(I,/obj/item/mule_module))
+		apply_module(I,user)
+		return
+	. = ..()
+
+/mob/living/simple_animal/mule_bot/welder_act(mob/living/user, obj/item/I)
+	var/repair_time = 1 SECONDS
+	if(src == user)
+		repair_time *= 3
+
+	user.visible_message(span_notice("[user] starts to fix some of the dents on [src]."),\
+		span_notice("You start fixing some of the dents on [src]"))
+
+	add_overlay(GLOB.welding_sparks)
+	while(do_after(user, repair_time, TRUE, src, BUSY_ICON_BUILD) && I.use_tool(volume = 50, amount = 2))
+		user.visible_message(span_warning("\The [user] patches some dents on [src]."), \
+			span_warning("You patch some dents on \the [src]."))
+		if(src.heal_limb_damage(15,15, robo_repair = FALSE, updating_health = TRUE))
+			UpdateDamageIcon()
+		if(!I.tool_use_check(user, 2))
+			break
+		if(!src.bruteloss)
+			balloon_alert(user, "Dents fully repaired.")
+			break
+	cut_overlay(GLOB.welding_sparks)
+	return TRUE
+
+/mob/living/simple_animal/mule_bot/crowbar_act(mob/living/user, obj/item/I)
+	remove_module()
+	return TRUE
+
+
+/mob/living/simple_animal/mule_bot/proc/swap_module(obj/item/mule_module/mod, mob/user)
+	remove_module(user, FALSE)
+	apply_module(mod,user, FALSE)
+	update_icon()
+
+/mob/living/simple_animal/mule_bot/proc/apply_module(obj/item/mule_module/mod, mob/user, update_icon = TRUE)
+	if(mod.apply(src))
+		to_chat(user,span_notice("You succesfully installed [mod]"))
+		user?.temporarilyRemoveItemFromInventory(mod)
+	else
+		to_chat(user,span_warning("You failed to installed [mod]"))
+	if(update_icon)
+		update_icon()
+
+/mob/living/simple_animal/mule_bot/proc/remove_module(mob/user, update_icon = TRUE)
+	if(!installed_module)
+		return
+	if(user)
+		to_chat(user,span_notice("You succesfully remove [installed_module]"))
+	installed_module.unapply(FALSE)
+	if(update_icon)
+		update_icon()
+
+
+/obj/item/remote
+	name = "remote"
+	icon = 'icons/obj/items/items.dmi'
+	icon_state = "multitool2"
+	var/mob/living/simple_animal/mule_bot/bot
+
+/obj/item/remote/attack_self(mob/user)
+	if(!bot)
+		return
+	if(SEND_SIGNAL(src, COMSIG_REMOTECONTROLL_STOP_FOLLOW, usr) & COMSIG_BOT_STOP)
+		to_chat(user,span_notice("[bot] will now stop following"))
```

Comment:
Also throughout, try to use balloon alerts instead of tochat
if you dont need to know the information 5 seconds after it's shown or it's represented visually already it just makes information less clear to the player